lib/linguist/languages.yml

```diff
@@ -3309,6 +3309,17 @@ ObjDump:
   tm_scope: objdump.x86asm
   ace_mode: assembly_x86
   language_id: 256
+ObjectScript:
+  type: programming
+  extensions:
+  - ".cls"
+  - ".inc"
+  - ".int"
+  - ".mac"
+  - ".xml"
```

Comment:
I would not add XML if it would be without any effect. I have added heuristic for XML, and test for it. I have a private repository, here, with sources in XML format, and tested linguist locally on this project as well, and it recognizes it exactly as I expected.

The result I got with all extensions
```
66.42%  ObjectScript
```
and without XML in `heuristics.yml`
```
45.13%  ObjectScript
```

Reply:
I'm not questioning your implementation. I'm saying that this is really not necessary as there's nothing specific about these XML files that makes them specifically ObjectScript as opposed to XML.

We already have 103 extensions associated with XML for various other "languages" which don't implement anything other than standard XML in those files. These files are no different and is confirmed by your use of the `.xml` extension.

Reply:
Additionally, if you identify these XML files as ObjectScript, the ObjectScript grammar will be used to perform the syntax highlighting, which is most definitely not what you want.

If you want the XML files you appear in your language statistics, you'll need to implement the "detectable" override.

Reply:
For us, XML is still source code, and I want to be able to search over it as well as any other code. When I will use search and limit it by Language ObjectScript, it should show results in any extensions, not only between "popular" amongst GitHub. As this language not as wide as any others, nobody has any possibility to find our source code by Language.

So, I need a way, when I can see Right statistics of languages in any repository. And the ability to search any ObjectScript extension in any repository.

> you'll need to implement the "detectable" override.
Could you please explain, how can I do it?

Reply:
> For us, XML is still source code, and I want to be able to search over it as well as any other code. When I will use search and limit it by Language ObjectScript, it should show results in any extensions, not only between "popular" amongst GitHub.  As this language not as wide as any others, nobody has any possibility to find our source code by Language.

You're not alone here and we even have an issue tracking all the other languages which have had PRs rejected due to poor popularity: https://github.com/github/linguist/issues/4219

Adding each and every language as it appears on the horizon adds to the overhead of running Linguist and could affect it's performance, hence we require a high degree of popularity before implementing support.

It would also be incorrect to do things like label XML files as another language by default as they're _not_ written in that language and would affect everyone on GitHub.com. The syntax highlighting would be a mess too.

Keep in mind, and as stated in the README, Linguist works by identifying the language of each file in isolation and uses this to present a breakdown of the languages within a repository. It does not identify a repository language as a whole.

> > you'll need to implement the "detectable" override.
> 
> Could you please explain, how can I do it?

https://github.com/github/linguist/blob/master/README.md#detectable has all the details.

If you then want to be treated these XML files as another language, you'd add an additional line indicating the language as per https://github.com/github/linguist/blob/master/README.md#using-gitattributes but remember this will cause those files to be interpreted as the new language and not XML and thus the specified language's grammar will be used to perform syntax highlighting and not the XML grammar.  Note, you _can't_ use it to force a language that Linguist doesn't know about.

Note, Linguist has no effect on searchability. _All_ files within a repo are indexed and searchable regardless of what Linguist knows about, provided they're not affected by the [code search restrictions](https://help.github.com/en/articles/searching-code#considerations-for-code-search). This is why you're able to find the files you linked to in your search results.